### PR TITLE
Settings/implement settings screen UI

### PR DESCRIPTION
**Pull request by Harrish92**
# Implementation of SettingsScreen UI and tests

## Description
This PR introduces the `SettingsScreen` components and their corresponding unit tests. It closes issue #218 .

## Changes
- Implemented `SettingsScreen` composable function with sections for notifications, themes, and account management.
- Added `DeleteAccountDialog` composable function for account deletion confirmation.
- Included detailed comments for each section and function to describe their purpose and usage.
- Added unit tests to verify the display and functionality of all components in the `SettingsScreen`.

#### Added
- `app/src/main/java/com/android/periodpals/ui/settings/SettingsScreen.kt`
- `app/src/main/java/com/android/periodpals/ui/components/SettingsComponents.kt`
- `app/src/main/java/com/android/periodpals/ui/settings/SettingsScreenTest.kt`

#### Modified
- `app/src/main/java/com/android/periodpals/resources/C.kt`
- `app/src/main/java/com/android/periodpals/ui/navigation/NavigationActions.kt`

#### Removed
- None

## Dependencies Added
- None

## Testing
- Verified that all components in the `SettingsScreen` are correctly displayed and interact as expected.
- Ensured that the delete account confirmation dialog is visible and functional.

## Screenshots
- See #218

### CODE REVIEW

File: app/src/main/java/com/android/periodpals/resources/Color.kt
Comment (francelu) [inline app/src/main/java/com/android/periodpals/resources/Color.kt:1]: Good ! Consistent with existing implementation.

File: app/src/main/java/com/android/periodpals/ui/components/SettingsComponents.kt
Comment (francelu) [inline app/src/main/java/com/android/periodpals/ui/components/SettingsComponents.kt:1]: Unsure whether this file is very relevant since this package is for shared components **across** screens (yeah I agree it is unclear in the package name). I would suggest just having these "shared" components directly inside "Settings.kt" (you named it `SettingsScreen.kt`) as it is easier to follow and read the code. Better practice, easier code maintenance and readability.

File: app/src/main/java/com/android/periodpals/ui/settings/SettingsScreen.kt [hunk 1]
```diff
@@ -2,33 +2,131 @@ package com.android.periodpals.ui.settings
...
   Scaffold(
-      modifier = Modifier.fillMaxSize(),
+      modifier = Modifier.fillMaxSize().testTag(SettingsScreen.SCREEN),
       topBar = {
         TopAppBar(
             title = SCREEN_TITLE,
-            backButton = true,
-            onBackButtonClick = { navigationActions.goBack() })
+            true,
+            onBackButtonClick = { navigationActions.navigateTo(Screen.PROFILE) },
```
Comment (francelu) [inline app/src/main/java/com/android/periodpals/ui/settings/SettingsScreen.kt:125]: Why not just use the `goBack()` function ?

File: app/src/main/java/com/android/periodpals/ui/components/SettingsComponents.kt [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+ * @param testTag the test tag for the description.
+ */
+@Composable
+fun SettingsDescription(text: String, testTag: String) {
+  Box(modifier = Modifier.fillMaxWidth()) {
+    Text(
+        text,
+        textAlign = TextAlign.Start,
+        style = MaterialTheme.typography.labelMedium,
+        modifier = Modifier.fillMaxWidth().wrapContentHeight().testTag(testTag),
```
Comment (francelu) [inline app/src/main/java/com/android/periodpals/ui/components/SettingsComponents.kt:64]: This should `.fillMaxWidth().wrapContentHeight()`

File: app/src/main/java/com/android/periodpals/ui/components/SettingsComponents.kt [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+ */
+@Composable
+fun SettingsSwitchRow(
+    text: String,
+    isChecked: Boolean,
+    onCheckedChange: (Boolean) -> Unit,
+    textTestTag: String,
+    switchTestTag: String
+) {
+  Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
```
Comment (francelu) [inline app/src/main/java/com/android/periodpals/ui/components/SettingsComponents.kt:87]: Add `.wrapContentHeight()` to the modifier. In general, for padding purposes, we fill the maximum width and wrap the height as we might have more elements than the available screen height. :)

File: app/src/main/java/com/android/periodpals/ui/settings/SettingsScreen.kt [hunk 4]
```diff
@@ -43,8 +141,201 @@ fun SettingsScreen(navigationActions: NavigationActions) {
...
+            textTestTag = SettingsScreen.TAMPONS_TEXT,
+            switchTestTag = SettingsScreen.TAMPONS_SWITCH)
+        SettingsDescription(COMMENT_ORGANIC, SettingsScreen.ORGANIC_DESCRIPTION)
+        SettingsSwitchRow(
+            text = NOTIF_ORGANIC,
+            isChecked = receiveNotifications && organicNotifications,
+            onCheckedChange = { organicNotifications = it },
+            textTestTag = SettingsScreen.ORGANIC_TEXT,
+            switchTestTag = SettingsScreen.ORGANIC_SWITCH)
+      }
```
Comment (francelu) [inline app/src/main/java/com/android/periodpals/ui/settings/SettingsScreen.kt:178]: Very good and clean implementation thanks to the extracted helper functions !! Lovely :)  
As mentioned before, having these helper functions at the end of this file would make more sense to me.

File: app/src/main/java/com/android/periodpals/ui/settings/SettingsScreen.kt [hunk 5]
```diff
@@ -43,8 +141,201 @@ fun SettingsScreen(navigationActions: NavigationActions) {
...
+            modifier = Modifier.testTag(SettingsScreen.THEME_DROP_DOWN_MENU_BOX),
+            expanded = expanded,
+            onExpandedChange = { expanded = it },
+        ) {
+          TextField(
+              modifier = Modifier.menuAnchor().fillMaxWidth(),
+              textStyle = MaterialTheme.typography.labelLarge,
+              value = theme,
+              onValueChange = {},
+              label = { Text(THEME_LABEL, style = MaterialTheme.typography.labelSmall) },
```
Comment (francelu) [inline app/src/main/java/com/android/periodpals/ui/settings/SettingsScreen.kt:192]: I think in general in other UIs, we used `labelMedium` for these kind of labels.  `labelSmall` is mainly used in the bottom navigation as it require a quite small font size text.

File: app/src/main/java/com/android/periodpals/ui/settings/SettingsScreen.kt [hunk 6]
```diff
@@ -43,8 +141,201 @@ fun SettingsScreen(navigationActions: NavigationActions) {
...
+ * @param onDismiss The callback to dismiss the dialog.
+ */
+@Composable
+private fun DeleteAccountDialog(navigationActions: NavigationActions, onDismiss: () -> Unit) {
+  Dialog(
+      onDismissRequest = onDismiss,
+      properties = DialogProperties(usePlatformDefaultWidth = false)) {
+        Card(
+            modifier =
+                Modifier.fillMaxWidth()
```
Comment (francelu) [inline app/src/main/java/com/android/periodpals/ui/settings/SettingsScreen.kt:278]: Add wrap height too. :)

File: app/src/main/java/com/android/periodpals/ui/navigation/NavigationActions.kt [hunk 7]
```diff
@@ -15,6 +15,7 @@ object Route {
   const val ALERT = "Alert"
   const val ALERT_LIST = "Alert List"
   const val MAP = "Map"
+  const val SETTINGS = "Settings"
```
Comment (francelu) [inline app/src/main/java/com/android/periodpals/ui/navigation/NavigationActions.kt:18]: This shouldn't be added here. Revert the changes.  
Why it shouldn't, because Settings is part of the *route* of profile. :) dunno if that's very clear. Basically, a route is a "starting point" from bottom nav menu (as I understand) beside auth which is the starting point of the app.

File: app/src/test/java/com/android/periodpals/ui/settings/SettingsScreenTest.kt [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+  private lateinit var navigationActions: NavigationActions
+  private lateinit var userViewModel: UserViewModel
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @Before
+  fun setUp() {
+    navigationActions = mock(NavigationActions::class.java)
+    userViewModel = mock(UserViewModel::class.java)
+
+    `when`(navigationActions.currentRoute()).thenReturn(Route.SETTINGS)
```
Comment (francelu) [inline app/src/test/java/com/android/periodpals/ui/settings/SettingsScreenTest.kt:38]: After removing `Route.SETTINGS`, change here with `Screen.SETTINGS`.

File: app/src/test/java/com/android/periodpals/ui/settings/SettingsScreenTest.kt [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+    composeTestRule.setContent { SettingsScreen(userViewModel, navigationActions) }
+
+    composeTestRule.onNodeWithTag(SettingsScreen.SCREEN).assertIsDisplayed()
+    composeTestRule.onNodeWithTag(TopAppBar.TOP_BAR).assertIsDisplayed()
+    composeTestRule
+        .onNodeWithTag(TopAppBar.TITLE_TEXT)
+        .assertIsDisplayed()
+        .assertTextEquals("My Settings")
+    composeTestRule.onNodeWithTag(TopAppBar.GO_BACK_BUTTON).assertIsDisplayed()
+    composeTestRule.onNodeWithTag(TopAppBar.EDIT_BUTTON).assertIsNotDisplayed()
```
Comment (francelu) [inline app/src/test/java/com/android/periodpals/ui/settings/SettingsScreenTest.kt:52]: Add the check for the not displayed settings button from the new top app bar.

File: app/src/test/java/com/android/periodpals/ui/settings/SettingsScreenTest.kt
Comment (francelu) [inline app/src/test/java/com/android/periodpals/ui/settings/SettingsScreenTest.kt:1]: Very good. Tests are complete !

File: app/src/main/java/com/android/periodpals/ui/settings/SettingsScreen.kt [hunk 10]
```diff
@@ -2,33 +2,131 @@ package com.android.periodpals.ui.settings
...
+private val THEME_DROPDOWN_CHOICES =
+    listOf(
+        listOf(THEME_SYSTEM, Icons.Outlined.PhoneAndroid),
+        listOf(THEME_LIGHT, Icons.Outlined.LightMode),
+        listOf(THEME_DARK, Icons.Outlined.DarkMode))
+
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
-fun SettingsScreen(navigationActions: NavigationActions) {
+fun SettingsScreen(userViewModel: UserViewModel, navigationActions: NavigationActions) {
```
Comment (francelu) [inline app/src/main/java/com/android/periodpals/ui/settings/SettingsScreen.kt:98]: Missing doc.

**Review by francelu — APPROVED** [review]
## Summary
Very good and clean implementation of the settings UI. Fit to Figma !

## Important
#### Code Quality
- Good in general.
- Could put all the helper functions from `SettingsComponents.kt` into `Settings.kt` directly as that's the only screen where the components are called. Make it far to reach and read code in an individual file.
- Code is consistent with rest of code.]

#### Functionality
- Works well !
- Well linked with other screens back and forth.

#### Performance
- Unchanged

#### Security
- Unchanged

## Testing
- Thorough, good and complete !

## Documentation
- Missing a documentation for `SettingsScreen` component.

## Suggestions
- See code comments.

## Steps before PR **merged**
- Check suggestions and reply / resolve.

**Review by taghizadlaura — APPROVED** [review]
## Summary
LGTM :
This PR implements the components of the `SettingsScreen` and the corresponding unit tests to ensure their proper functionality. 

## Important

#### Code Quality
LGTM :
- The code is well-organized with appropriate comments explaining the components and their respective functions.
- The `SettingsScreen` composable is clean, and its sections are logically divided for notifications, themes, and account management.

#### Functionality
LGTM :
- The functionality seems well-implemented for displaying the `SettingsScreen` with all components like notifications, themes, and account management sections.

#### Performance
- No performance issues.

#### Security
LGTM :
- The account deletion functionality is handled correctly, ensuring that proper confirmation is in place to prevent unintentional deletions.

## Testing
LGTM :
The testing thoroughly covers the functionality of the `SettingsScreen`, ensuring all components are displayed, navigational actions are correct, and user interactions, including account deletion, are well-handled.

## Documentation
LGTM :
- Functions are well-documented in general, providing clarity for future development. 
- The `SettingsScreen` function could benefit from additional documentation to improve understanding of its structure and purpose.

## Suggestions
- Add documentation for `SettingsScreen`.
- Answer my suggestion about the signout color.

## Steps before PR approved
- Respond to my suggestions.
- Address France's comments (I didn’t repeat what she mentioned, but I agree with her).
